### PR TITLE
Anonymous User: Adds validator service for anonymous users

### PR DESCRIPTION
**Pull request by leandro-deveikis**
Fixes https://github.com/grafana/grafana-operator-experience-squad/issues/1017

### CODE REVIEW

**Review by IevaVasiljeva — APPROVED** [review]
The approach looks good to me 👌

File: pkg/services/anonymous/anonimpl/impl.go [hunk 1]
```diff
@@ -81,6 +85,11 @@ func (a *AnonDeviceService) usageStatFn(ctx context.Context) (map[string]any, er
 }
 
 func (a *AnonDeviceService) tagDeviceUI(ctx context.Context, device *anonstore.Device) error {
+	err := a.validator.Validate(ctx)
```
Comment (eleijonmarck) [inline pkg/services/anonymous/anonimpl/impl.go:88]: @leandro-deveikis  nice work.

i find that validator/validate is a bit abstract, it is enforcing a limit. so should the interface be `CheckLimit` .

to me i read it as, validate what? :honk_think:

**Review by eleijonmarck — CHANGES_REQUESTED** [review]

File: pkg/services/anonymous/anonimpl/impl.go (hunk 1 above)
Comment (leandro-deveikis) [reply pkg/services/anonymous/anonimpl/impl.go]: fixed in last commit! thanks for the feedback!

**Review by eleijonmarck — APPROVED** [review]
explicit naming for now 👍

**Pull request merged**